doc/guide/dynamics/dynamics-time.rst

```diff
@@ -270,38 +270,45 @@ Collapse operators are handled in the exact same way.
 Modeling Non-Analytic and/or Experimental Time-Dependent Parameters using Interpolating Functions
 =================================================================================================
 
-Sometimes it is necessary to model a system where the time-dependent parameters are non-analytic functions, or are derived from experimental data (i.e. a collection of data points).  In these situations, one can use interpolating functions as an approximate functional form for input into a time-dependent solver.  QuTiP includes it own custom cubic spline interpolation class :class:`qutip.interpolate.Cubic_Spline` to provide this functionality.  To see how this works, lets first generate some noisy data:
+Sometimes it is necessary to model a system where the time-dependent parameters are non-analytic functions, or are derived from experimental data (i.e. a collection of data points).
+In these situations, one can use interpolating functions as an approximate functional form for input into a time-dependent solver.
+QuTiP support spline interpolation in it's :class:`qutip.Coefficient`.
+To see how this works, lets first generate some noisy data:
 
 .. plot::
     :context:
 
-    t = np.linspace(-15, 15, 100)
-    func = lambda t: 9*np.exp(-(t / 5)** 2)
-    noisy_func = lambda t: func(t)+(0.05*func(t))*np.random.randn(t.shape[0])
-    noisy_data = noisy_func(t)
+    times = np.linspace(-15, 15, 100)
+    func = lambda t: 9 * np.exp(-(t / 5)** 2)
+    noisy_data = np.array([
+        func(t) * (1 + 0.05 * np.random.randn())
+        for t in times
+    ])
```

Comment:
Could we not keep this vectorised as `noisy_data = func(times) * (1 + 0.05 * np.random.randn(len(times)))`?